providers/provider_data.go

```diff
@@ -20,24 +21,42 @@ const (
 	OIDCGroupsClaim = "groups"
 )
 
+// LoginURLParameter is the configuration for a single query parameter that
+// can be passed through from the /oauth2/start endpoint to the IdP login
+// URL.  This is a parallel struct to the one in the options package, but
+// with the regexp pre-compiled.
+type LoginURLParameter struct {
```

Comment:
Is this not the same as the one in `options`? Should we not import from there, does that cause a circular dependency?

Reply:
The two structs are similar but not quite the same, the one in options has a `*string` for the regular expression, the one in here holds a pointer to the `regexp.Regexp` that was compiled from that string.